File: 26. Remove Duplicates from Sorted Array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.size() < 2) return nums.size();
        int res, left, right;
        res = right = 1, left = 0;
        while (right < nums.size()) {
            if (nums[right] != nums[left]) {
                nums[++left] = nums[right++];
                res++;
            } else {
                while (right < nums.size() && nums[right] == nums[left]) {
                    right++;
                }
            }
        }
        return res;
    }
};

File: java/26. Remove Duplicates from Sorted Array.java
// O(n)
class Solution {
    public int removeDuplicates(int[] nums) {
        int i = 0, j = 0;
        while (j < nums.length) {
            if (j == 0 || (nums[j] != nums[j - 1])) {
                nums[i++] = nums[j];
            }
            ++j;
        }
        return i;
    }
}
